docarray/array/storage/milvus/backend.py

```diff
@@ -0,0 +1,320 @@
+        config_subindex: dict,
+        config_joined: dict,
+        subindex_name: str,
+    ) -> dict:
+        if 'collection_name' not in config_subindex:
+            config_joined['collection_name'] = _sanitize_collection_name(
+                config_joined['collection_name'] + '_subindex_' + subindex_name
+            )
+        return config_joined
+
+    def _doc_to_milvus_payload(self, doc):
+        return self._docs_to_milvus_payload([doc])
+
+    def _docs_to_milvus_payload(self, docs: 'Iterable[Document]'):
+        extra_columns = [
+            [self._map_column(doc.tags.get(col), col_type) for doc in docs]
+            for col, col_type in self._config.columns.items()
+        ]
+        return [
+            [doc.id for doc in docs],
+            [self._map_embedding(doc.embedding) for doc in docs],
+            [doc.to_base64(**self._config.serialize_config) for doc in docs],
+            *extra_columns,
+        ]
+
+    @staticmethod
+    def _docs_from_query_response(response):
+        return DocumentArray([Document.from_base64(d['serialized']) for d in response])
+
+    @staticmethod
+    def _docs_from_search_response(
+        responses,
+    ) -> 'Union[List[DocumentArray], DocumentArray]':
+        das = []
+        for r in responses:
+            das.append(
+                DocumentArray(
+                    [Document.from_base64(hit.entity.get('serialized')) for hit in r]
+                )
+            )
+        return das if len(das) > 0 else das[0]
+
+    def _update_consistency_level(self, **kwargs):
+        kwargs_consistency_level = kwargs.get('consistency_level', None)
+        kwargs['consistency_level'] = (
+            kwargs_consistency_level
+            if kwargs_consistency_level
+            else self._config.consistency_level
+        )
+        return kwargs
+
+    def _map_embedding(self, embedding):
+        if embedding is not None:
+            from docarray.math.ndarray import to_numpy_array
+
+            embedding = to_numpy_array(embedding)
+
+            if embedding.ndim > 1:
+                embedding = np.asarray(embedding).squeeze()
+        else:
+            embedding = np.zeros(self._config.n_dim)
+        return embedding
+
+    def __getstate__(self):
+        d = dict(self.__dict__)
+        del d['_collection']
+        del d['_offset2id_collection']
+        return d
+
+    def __setstate__(self, state):
+        self.__dict__ = state
+        self._collection = self._create_or_reuse_collection()
+        self._offset2id_collection = self._create_or_reuse_offset2id_collection()
+
+    def __enter__(self):
+        _ = super().__enter__()
+        self._collection.load()
+        self._offset2id_collection.load()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self._collection.release()
+        self._offset2id_collection.release()
+        super().__exit__(exc_type, exc_val, exc_tb)
+
+    def loaded_collection(self, collection=None):
```

Comment:
it looks like this is used in our implemented operations + loading/releasing is offered using DA context manager.
Is the user supposed to use this in cases other than the DA context manager ?
If there are no such cases, I suggest we make this private

Reply:
The DA context manager also handled offset2id, and could potentially do other stuff in the future. I think it can be useful to expose this if the user wants to make sure that nothing other than loading/releasing is happening.

Reply:
I would use a simple context manager, I think is great value to have the simplest the better

Reply:
then let's document this as part of the public interface in the Collection loading section